tests/Constraints/SelfDefinedSchemaTest.php

```diff
@@ -71,6 +71,6 @@ public function testInvalidArgumentException()
     {
         $v = new Validator();
         $this->setExpectedException('\JsonSchema\Exception\InvalidArgumentException');
-        $v->check(json_decode('{}'), json_decode(''));
+        $v->validate(json_decode('{}'), json_decode(''));
```

Comment:
You cannot use `validate()` in this situation, as the first argument is a reference. If you want to change this, you'll need to decode the document into an intermediate variable first, and then pass the variable.